### PR TITLE
Localization Support For In Game Weapons & Items

### PR DESCRIPTION
**Pull request by NovaDiablox**
I modified @unknown-gd's work. All credit goes to him. Without him this wouldn't be possible. You can close https://github.com/Facepunch/garrysmod/pull/2061.
image

### CODE REVIEW

File: garrysmod/lua/autorun/game_hl2.lua [hunk 1]
```diff
@@ -1,79 +1,112 @@
...
-local function ADD_ITEM( name, class, offset, extras, classOverride )
-
-	local base = { PrintName = name, ClassName = class, Category = Category, NormalOffset = offset or 32, DropToFloor = true, Author = "VALVe" }
-	list.Set( "SpawnableEntities", classOverride or class, table.Merge( base, extras or {} ) )
-	duplicator.Allow( class )
-
+local spawnableEntities = list.GetForEdit( "SpawnableEntities" )
+local npcWeapons = list.GetForEdit( "NPCUsableWeapons" )
+local playerWeapons = list.GetForEdit( "Weapon" )
+local Allow = duplicator.Allow
```
Comment (robotboy655) [inline garrysmod/lua/autorun/game_hl2.lua:4]: I can tell you right now, you do not need to do this silly stuff here.

Comment (NovaDiablox) [reply garrysmod/lua/autorun/game_hl2.lua:4]: I don't know coding and tried to modify unknowns code and trying to fix rn.

**Comment by NovaDiablox** [comment]
Well I found the problem will update it soon.

**Comment by NovaDiablox** [comment]
image
As you can see it is working, if the Turkish string doesn't exist it falls back to English.

**Comment by NovaDiablox** [comment]
image
image

**Comment by NovaDiablox** [comment]
I think it's ready.

**Comment by NovaDiablox** [comment]
I just tested it in 'dev' branch and it works fine. Previous tests were made in x86-64 branch.
image
image
image
image

**Comment by NovaDiablox** [comment]
@robotboy655 It is ready.

**Comment by NovaDiablox** [comment]
I got permission from him if you have ownership concerns.
image

**Review by robotboy655 — COMMENTED** [review]
While I appreciate the idea, I fundamentally disagree with how you went about implementing it.

First of all, you chose to reuse existing translations for certain things, and while I can maybe see what you were trying to do, this causes a few issues:
1) Modifying capitalization of existing strings (Weapon names in the weapon selection HUD), or the meaning of the string entirely. (All grenades would become "Zombine Grenades" because you merged the strings without thinking. The `npc_grenade_frag` class is reused throughout the game, you cannot simply name it "Zombine Grenade".)
2) There is no consistency with how entity names are localized, meaning certain entity names are localized (e.g. `#combine_mine`), but others are not. (e.g. `#item_ammo_357`)
3) Tying ammo localization strings to entity names is in itself flawed. This ties into point # 2, but there are cases where you'd want the ammo name to be different from the entity that gives you said ammo.

Secondly, the way you implement translations is quite wasteful, having essentially duplicated class names in the Lua files for each entity. This could be done better.

This is also a good opportunity to do a pass on what the names of each of these things are, and have the names be more representative of what the entity actually is. i.e. change `Pistol Ammo` to `Pistol Ammo Box` so the name better represents what the entity actually is.

With this being said, I have applied changes how I think this should be done, including changing some string around. This omits changes to ammo labels themselves as well as weapon names in weapon switching, since they are not part of the Lua file and can be done separately.

I have committed the changes to `dev` beta as well as a test (and because I basically rebuilt the entire PR, so the work is already done), but I will not merge this yet, let me know if you disagree on anything, and if not I will merge this pull request.

I have not included some of the other changes I made to this PR (basically removing the new strings added to .properties files from the .txt files where they used to be)

Here's a preview of the new strings in spawnmenu:
image

**Comment by NovaDiablox** [comment]
Thank you so much Rubat! Since you made the changes right, you may close this PR. If something is missing you can complete it. I completely agree with you on gameplay concept of pistol ammo and other stuff you mentioned. The preview also summarizes your changes. It looks good and refined.

**Comment by Satton2** [comment]
@robotboy655 

A huge thank you for your work and merging this.

Can we expect this to be added on Crowdin so we can localize it there, OR we should forget about Crowdin on this one and localize via GitHub?

**Pull request merged**